Src/WICTextureLoader.cpp

```diff
@@ -154,16 +164,11 @@ namespace DirectX
         static INIT_ONCE s_initOnce = INIT_ONCE_STATIC_INIT;
 
         IWICImagingFactory2* factory = nullptr;
-        (void)InitOnceExecuteOnce(&s_initOnce,
-            [](PINIT_ONCE, PVOID, PVOID *ifactory) -> BOOL
```

Comment:
So this builds fine with Visual C++ and with clang when using x64, but fails with clang on x86.

Turns out the problem is that clang doesn't implement casting stateless lambdas to different calling conventions, so this code fails.

Sadly, I refactored this into a classic callback function instead of a lambda...